src/Adapter/Feature/Repository/FeatureRepository.php

```diff
@@ -53,16 +58,69 @@ class FeatureRepository extends AbstractObjectModelRepository
      */
     protected $dbPrefix;
 
+    /**
+     * @var FeatureValidator
+     */
+    private $featureValidator;
+
     /**
      * @param Connection $connection
      * @param string $dbPrefix
+     * @param FeatureValidator $featureValidator
      */
     public function __construct(
         Connection $connection,
-        string $dbPrefix
+        string $dbPrefix,
+        FeatureValidator $featureValidator
     ) {
         $this->connection = $connection;
         $this->dbPrefix = $dbPrefix;
+        $this->featureValidator = $featureValidator;
+    }
+
+    public function get(FeatureId $featureId): Feature
+    {
+        /** @var Feature $feature */
+        $feature = $this->getObjectModel(
+            $featureId->getValue(),
+            Feature::class,
+            FeatureNotFoundException::class
+        );
+
+        return $feature;
+    }
+
+    /**
+     * @param array<int, string> $localizedNames
+     * @param ShopId[] $associatedShopIds
+     *
+     * @return Feature
+     */
+    public function create(
+        array $localizedNames,
+        array $associatedShopIds
+    ): Feature {
+        $feature = new Feature();
+        $feature->name = $localizedNames;
+        $feature->id_shop_list = array_map(static function (ShopId $shopId) {
+            return $shopId->getValue();
+        }, $associatedShopIds);
```

Comment:
You should use `AbstractMultiShopObjectModelRepository::addObjectModelToShops` instead which does exactly this